### PR TITLE
Improves the syndicate teleporter

### PR DESCRIPTION
**Pull request by Cenrus**
## About The Pull Request

Added num2loadingbar text helper from tg station (See below)

tele

Added a loading bar to the teleporter making it easier to identify when it is about to recharge, and a little sound effect when that happens.

Cooldown changed from 20 seconds to 15 seconds

Commented on some variables in teleportation.dm

Deleted the admin only teleporter with 999 charges, it is useless and can be recreated with var editing

## Why It's Good For The Game

I noticed myself shift clicking this thing way too much when using it to see if it is actually able to teleport me, and I think I put the recharge time a little too high.

## Changelog
:cl: Cenrus, MrDoomBringer
balance: Reduced recharge time of the syndicate teleporter to 15 seconds (previously 20)
add: Added a recharge timer to the syndicate teleporter that is shown on examine
add: Added a sound to indicate that the teleporter has recharged
/:cl:

### CODE REVIEW

**Comment by utilbird** [comment]
damn cool helper

File: code/game/objects/items/teleportation.dm [hunk 1]
```diff
@@ -253,17 +253,29 @@
...
+	var/recharge_timer
 
 /obj/item/teleporter/examine(mob/user)
 	. = ..()
 	. += "<span class='notice'>[src] has [charges] out of [max_charges] charges left.</span>"
+	if(recharging)
+		. += "<span class='notice'><b>A small display on the back reads:</b></span>"
+		var/timeleft = timeleft(recharge_timer)
+		var/loadingbar = num2loadingbar(timeleft/recharge_time)
+		. += "<span class='notice'><b>CHARGING: [loadingbar] ([timeleft*0.1]s)</b></span>"
```
Comment (utilbird) [inline code/game/objects/items/teleportation.dm]:
```suggestion
		var/timeleft = timeleft(recharge_timer)
		var/loadingbar = num2loadingbar(timeleft/recharge_time)
		. += "<span class='notice'><b>CHARGING: [loadingbar] ([timeleft SECONDS]s)</b></span>"
```

File: code/__HELPERS/text.dm [hunk 2]
```diff
@@ -840,3 +840,10 @@ GLOBAL_LIST_INIT(binary, list("0","1"))
 			continue
 		out += prob(replaceprob) ? pick(replacementchars) : char
 	return out.Join("")
+
+/proc/num2loadingbar(percent as num, numSquares = 20, reverse = FALSE)
+	var/loadstring = ""
+	for (var/i in 1 to numSquares)
+		var/limit = reverse ? numSquares - percent*numSquares : percent*numSquares
```
Comment (utilbird) [inline code/__HELPERS/text.dm]: does the limit need to be redefined every square? you should just be able to have it below loadstring definition for more performance

**Comment by Cenrus** [comment]
yeah

**Review by utilbird — APPROVED** [review]

**Pull request merged**